packages/react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -382,48 +559,49 @@ const ComboBox = React.forwardRef((props, ref) => {
                   translateWithId={translateWithId}
                 />
               </div>
-              <ListBox.Menu {...getMenuProps({ 'aria-label': ariaLabel })}>
+              <ListBoxMenu {...getMenuProps({ 'aria-label': ariaLabel })}>
                 {isOpen
                   ? filterItems(items, itemToString, inputValue).map(
                       (item, index) => {
                         const itemProps = getItemProps({
                           item,
                           index,
-                          ['aria-current']: selectedItem === item ? true : null,
+                          ['aria-current']: selectedItem === item ? "true" : "false",
                           ['aria-selected']:
-                            highlightedIndex === index ? true : null,
+                            highlightedIndex === index ? "true" : "false",
                           disabled: item.disabled,
                         });
                         return (
-                          <ListBox.MenuItem
+                          <ListBoxMenuItem
                             key={itemProps.id}
                             isActive={selectedItem === item}
                             isHighlighted={
                               highlightedIndex === index ||
-                              (selectedItem?.id &&
-                                selectedItem?.id === item.id) ||
+                              ((selectedItem as any)?.id &&
+                              (selectedItem as any)?.id === item.id) ||
                               false
                             }
                             title={
-                              itemToElement ? item.text : itemToString(item)
+                              itemToElement ? item.text : itemToString ? itemToString(item) : defaultItemToString(item)
```

Comment:
Why is the addition of a new `default*()` function necessary here?

Reply:
`itemToString` is possibly undefined, so we needed to check; but looking at this again, I'm not sure the default is really what we want here, since we don't know what item is or what properties it has.

Reply:
Updated to be 
```
                         title={
                              itemToElement
                                ? item.text
                                : itemToString
                                ? itemToString(item)
                                : undefined
                            }
``` 
which will keep the original behavior